model_analyzer/result/result_manager.py

```diff
@@ -480,133 +487,195 @@ def _tabulate_measurements(self, run_config_result):
-            for metric in metric_list:
-                metric_tag_index = self._find_index_for_field(
-                    inference_fields, metric.tag)
-
-                if metric_tag_index is not None:
-                    inference_row[metric_tag_index] = round(metric.value(), 1)
+        self._populate_inference_rows(run_config_measurement, inference_fields,
+                                      inference_row)
 
         self._result_tables[self.model_inference_table_key].insert_row_by_index(
             inference_row)
 
         # GPU specific data (only put measurement if not cpu only)
-        if not cpu_only:
+        if not any(cpu_onlys):
             for gpu_uuid, metrics in run_config_measurement.gpu_data().items():
                 gpu_fields = self._gpu_output_fields
-                gpu_row = self._get_common_row_items(
-                    gpu_fields, batch_size, concurrency, satisfies, model_name,
-                    model_config_name, dynamic_batching, instance_group)
-                gpu_uuid_index = self._find_index_for_field(
-                    gpu_fields, 'gpu_uuid')
-                if gpu_uuid_index is not None:
-                    gpu_row[gpu_uuid_index] = gpu_uuid
-                for metric in metrics:
-                    metric_tag_index = self._find_index_for_field(
-                        gpu_fields, metric.tag)
-                    if metric_tag_index is not None:
-                        gpu_row[metric_tag_index] = round(metric.value(), 1)
+
+                gpu_row = self._get_common_row_items(gpu_fields, batch_sizes,
+                                                     concurrencies, satisfies,
+                                                     model_name,
+                                                     model_config_name,
+                                                     dynamic_batchings,
+                                                     instance_groups)
+
+                self._add_uuid_to_gpu_row(gpu_row, gpu_uuid, gpu_fields)
+                self._add_metrics_to_gpu_row(gpu_row, metrics, gpu_fields)
+
                 self._result_tables[
                     self.model_gpu_table_key].insert_row_by_index(row=gpu_row)
 
+    def _tabulate_measurement_setup(self, run_config_measurement):
+        model_specific_pa_params = run_config_measurement.model_specific_pa_params(
+        )
+        batch_sizes = [
+            pa_params['batch-size'] for pa_params in model_specific_pa_params
+        ]
+        concurrencies = [
+            pa_params['concurrency-range']
+            for pa_params in model_specific_pa_params
+        ]
+
+        return model_specific_pa_params, batch_sizes, concurrencies
+
+    def _populate_inference_rows(self, run_config_measurement, inference_fields,
+                                 inference_row):
+        # Just using this to get a list of tags, so only need to reference the first model
```

Comment:
is it possible that models have different tags? If one model is cpu_only and the other isn't?

Reply:
Yes, probably. I'll file a story to figure out how we want to deal with this.

Reply:
TMA-686 filed